unwind/src/libunwind_shim.rs

```diff
@@ -107,6 +110,16 @@ pub unsafe extern "C" fn _Unwind_FindEnclosingFunction(pc: *mut c_void) -> *mut
     pc // FIXME: implement this
 }
 
+// FIXME: Set `unwind(allowed)` because we need to be able to unwind this function as
+// part of its operation, but we should ensure:
+// - it never needs any cleanup
+// - it never panics (this applies to all these extern functions, but especially here
+// now that we are specifying `unwind(allowed)`)
```

Comment:
As a sidenote, I would argue that "needing cleanup" is not a problem in principle, it's just terrible when the actual unwinder object lives on the stack (thus should actually not be cleaned up).

The thing about panicking is that the entire point of this file is to bridge to rust's unwinding code - written in Rust. So again I think panicking is not a problem in principle (although it may still be strictly UB and thus a footgun, this `unwind(allowed)` mess being the prime example). The much bigger issue I see is that we have to find a general solution for error handling in this crate that does not involve panicking at all, in any place, so we don't run into infinite loops.

Reply:
I agree. I've changed the comment.